### PR TITLE
Issue #355 Lifecycle concerns for persistent data

### PR DESCRIPTION
**Pull request by ljacomet**
Now controlled through LocalPersistenceService and PersistentCacheManager in place of Store level.

* Change LocalPersistenceService API
* Adapt DiskStore creation
* Change LocalPersistenceService API
* Wire destruction of persistence context into PersistentCacheManager.destroyCache
* Enable destruction of persistence context without binding a cache
* Deleted StoreDestroyTest as this is no longer a Store concern

This is the first step in issue #355 `UserManagedCache` still needs to be addressed.

### CODE REVIEW

**Comment by ljacomet** [comment]
Pushed updated commit with minor clean up.

**Comment by ljacomet** [comment]
Still not a single comment ... 
Can someone please merge then? Or ping me to do a drive through one more time?

**Comment by AbfrmBlr** [comment]
I understand that we have decoupled the lifecycle concern of cache with the persistent store(context) and now creation/destruction is taken care by the persistence service itself.

File: core/src/main/java/org/ehcache/Ehcache.java [hunk 1]
```diff
@@ -1075,28 +1074,12 @@ public CacheStatistics getStatistics() {
...
-    try {
-      if (store instanceof Persistable) {
-        logger.info("Creating Cache storage.");
-        ((Persistable) store).create();
-        logger.info("Cache storage successfully created.");
-      }
-    } catch (Exception e) {
-      throw new RuntimeException("Couldn't create Cache storage ", e);
-    }
+    // TODO figure out persistence and user managed caches
```
Comment (alexsnaps) [inline core/src/main/java/org/ehcache/Ehcache.java]: Where or when is that todo to be addressed? Is there an other issue for that?

Comment (AbfrmBlr) [inline core/src/main/java/org/ehcache/Ehcache.java]: If I recall it right , this is just the partial PR which talks about API stuff. There will be further PRs for the same issue

Comment (alexsnaps) [inline core/src/main/java/org/ehcache/Ehcache.java]: Duh! Y...

**Pull request merged**